include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,127 @@
+ *
+ * ```cpp
+ * // WRONG EXAMPLE:
+ * dna4 my_letter{0};      // we want to set the default, an A
+ * dna4 my_letter{'A'};    // we also want to set an A, but we are setting value 65
+ *
+ * std::cout << my_letter; // you expect 'A', but how would you access the number?
+ * ```
+ *
+ * To solve this problem, every alphabet defines two interfaces:
+ *
+ *   1. a **character based interface** with
+ *     1. the underlying character type able to represent this alphabet visually (almost always `char`,
+ * but could be `char16_t` or `char32_t`, as well)
+ *     2. a seqan3::to_char function to produce the visual representation
+ *     3. a seqan3::assign_char function to assign from the visual representation
+ *   2. a **rank based interface** with
+ *     1. the underlying rank type able to represent this alphabet numerically; this type must be able to represent
+ * the numbers from `0` to `alphabet size - 1` (often `uint8_t`, but sometimes a larger unsigned integral type)
+ *     2. a seqan3::to_rank function to produce the numerical representation
+ *     3. a seqan3::assign_rank function to assign from the numerical representation
+ *
+ * To prevent the aforementioned ambiguity, you can neither assign from rank or char representation via `operator=`,
+ * nor can you cast the alphabet to either of it's representation forms, **you need to explicitly use the
+ * interfaces**:
+ * ```cpp
+ * dna4 my_letter;
+ * assign_rank(my_letter, 0);       // assign an A via rank interface
+ * assign_char(my_letter, 'A');     // assign an A via char interface
+ * my_letter = dna4::A;             // some alphabets(BUT NOT ALL!) also provide an enum-like interface
+ *
+ * std::cout << to_char(my_letter);            // prints 'A'
+ * std::cout << (unsigned)to_rank(my_letter);  // prints 0
+ * // we have to add the cast here, because uint8_t is also treated as a char type by default :(
+ * ```
+ *
+ * For efficiency the representation saved internally is normally the rank representation, and the character
+ * representation
+ * is generated via conversion tables, however this is not required as long as both interfaces are provided
+ * and all functions operate in constant time.
+ *
+ * <small>In the documentation you will also encounter seqan3::semi_alphabet_concept. It describes "one half" of an
+ * alphabet and only defines the rank interface as a type requirement. It is mainly used internally and not
+ * relevant to most users of SeqAn.</small>
+ *
+ * ### members VS free/global functions
+ *
+ * The alphabet concept (as most concepts in SeqAn) looks for free/global functions, i.e. you need to be able
+ * to call `seqan3::to_rank(my_letter)`, however *most* alphabets also provide a member function, i.e.
+ * `my_letter.to_rank()`. The same is true for the metafunction seqan3::alphabet_size vs the static data member
+ * `value_size`.
+ *
+ * Members are provided for convenience and if you work with concrete types you are fine with using the member
```

Comment:
Mhmm, I would put the emphasis more on being safe and always use the free function. 
Only if one can be certain, that the concrete types are our alphabets with the member function one can use them.